pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -110,11 +110,13 @@ func newPGOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
 	}
 
 	isController := true
+	blockOwnerDeletion := true
 	return []metav1.OwnerReference{{
-		APIVersion: v1.SchemeGroupVersion.Version,
-		Kind:       "Pod",
-		Controller: &isController,
-		Name:       pod.Name,
-		UID:        pod.UID,
+		APIVersion:         v1.SchemeGroupVersion.Version,
+		Kind:               "Pod",
+		Controller:         &isController,
+		Name:               pod.Name,
+		UID:                pod.UID,
+		BlockOwnerDeletion: &blockOwnerDeletion,
```

Comment:
nit: Instead of constructing this one by one. Can use `metav1.NewControllerRef` function.

Reply:
Good suggestion! I've updated the PR.